pinecone/config/pinecone_config.py

```diff
@@ -1,12 +1,24 @@
 from typing import Optional, Dict
+import logging
+import json
 import os
 from .config import ConfigBuilder, Config
 
+logger = logging.getLogger(__name__)
+
 DEFAULT_CONTROLLER_HOST = "https://api.pinecone.io"
 
 
 class PineconeConfig():
     @staticmethod
     def build(api_key: Optional[str] = None, host: Optional[str] = None, additional_headers: Optional[Dict[str, str]] = {},  **kwargs) -> Config:
         host = host or kwargs.get("host") or os.getenv("PINECONE_CONTROLLER_HOST") or DEFAULT_CONTROLLER_HOST
+        headers_json = os.getenv("PINECONE_ADDITIONAL_HEADERS")
+        if headers_json:
+          try:
+            headers = json.loads(headers_json)
+            additional_headers = additional_headers or headers
+          except json.decoder.JSONDecodeError as e:
```

Comment:
I would probably catch all exceptions here just to be robust against any changes that might occur in how the `json` package structures their errors. We have to work with multiple python versions and future versions that could make changes here.

```suggestion
          except Exception as e:
```

Reply:
makes sense, will make the change